### PR TITLE
March 10 Showbuild

### PR DESCRIPTION
**Pull request by randomPoison**
- New level design.
- New art assets.
- New buddy model and animations.
- Night lights turn off during the day.

### CODE REVIEW

File: Assets/Scripts/Buddy/BuddyShaper.cs [hunk 1]
```diff
@@ -10,27 +11,58 @@ public class BuddyShaper : MonoBehaviour
...
 			skinnedMeshRend.SetBlendShapeWeight( i, setWeight );
-
+			
 			if( i == legIndex )
 			{
 				transform.position += Vector3.up * setWeight * legHeightOffsetMod;
 			}
 		}
-
+
```
Comment (randomPoison) [inline Assets/Scripts/Buddy/BuddyShaper.cs]: Trailing whitespace here and on line 38.

File: Assets/Scripts/Buddy/BuddyShaper.cs [hunk 2]
```diff
@@ -10,27 +11,58 @@ public class BuddyShaper : MonoBehaviour
...
 			{
 				transform.position += Vector3.up * setWeight * legHeightOffsetMod;
 			}
 		}
-
+		
 		int exclusiveBlendIndex = Random.Range( 0, exclusiveBlendIndices.Length );
 		skinnedMeshRend.SetBlendShapeWeight( exclusiveBlendIndices[exclusiveBlendIndex], 
-		                                     Random.Range( exclusiveBlendRange.min, exclusiveBlendRange.max ) );
+		                                    Random.Range( exclusiveBlendRange.min, exclusiveBlendRange.max ) );
```
Comment (randomPoison) [inline Assets/Scripts/Buddy/BuddyShaper.cs]: Add a test to make sure `exclusiveBlendIndicies` isn't empty or we could get an exception.

File: Assets/Scripts/Buddy/BuddyShaper.cs [hunk 3]
```diff
@@ -10,27 +11,58 @@ public class BuddyShaper : MonoBehaviour
...
+	}
+
+	void AdjustSize()
+	{
+		Vector3 adjustedScale = transform.localScale;
+		adjustedScale.y *= heightScaleRange.Random;
+
+		transform.localScale = adjustedScale;
```
Comment (randomPoison) [inline Assets/Scripts/Buddy/BuddyShaper.cs]: This could be done in one line with the `SetY()` extension method.

File: Assets/Scripts/Buddy/BuddyShaper.cs [hunk 4]
```diff
@@ -10,27 +11,58 @@ public class BuddyShaper : MonoBehaviour
...
+	}
+
+	void AdjustColor()
+	{
+		Color colorOffset = new Color( -Mathf.Clamp01( colorOffsetRange.Random ),
+		                              Mathf.Clamp01( colorOffsetRange.Random ),
+		                              Mathf.Clamp01( colorOffsetRange.Random ),
+		                              0f );
```
Comment (randomPoison) [inline Assets/Scripts/Buddy/BuddyShaper.cs]: These aren't lined up.

File: Assets/Scripts/Tags/NightLightTag.cs [hunk 5]
```diff
@@ -3,4 +3,34 @@
...
+		ResetLight();
+	}
+
+	void ResetLight()
+	{
+		_light.enabled = false;
+
+		// calculate time until the light
+		// needs to turn on again.
+		float timeToOn = DayCycleManager.dayCycleLength * _timeOn - DayCycleManager.currentTime;
```
Comment (randomPoison) [inline Assets/Scripts/Tags/NightLightTag.cs]: Add a case for if its already night time.

**Comment by randomPoison** [comment]
I'm done looking this over.

File: Assets/Scripts/Buddy/BuddyShaper.cs (hunk 1 above)
Comment (randomPoison) [inline Assets/Scripts/Buddy/BuddyShaper.cs]: Done.

File: Assets/Scripts/Buddy/BuddyShaper.cs (hunk 2 above)
Comment (randomPoison) [inline Assets/Scripts/Buddy/BuddyShaper.cs]: Done.

File: Assets/Scripts/Buddy/BuddyShaper.cs (hunk 3 above)
Comment (randomPoison) [inline Assets/Scripts/Buddy/BuddyShaper.cs]: Done.

File: Assets/Scripts/Buddy/BuddyShaper.cs (hunk 4 above)
Comment (randomPoison) [inline Assets/Scripts/Buddy/BuddyShaper.cs]: Done.

File: Assets/Scripts/Tags/NightLightTag.cs (hunk 5 above)
Comment (randomPoison) [inline Assets/Scripts/Tags/NightLightTag.cs]: Done.

File: Assets/Scripts/Buddy/BuddyShaper.cs [hunk 6]
```diff
@@ -10,27 +11,59 @@ public class BuddyShaper : MonoBehaviour
...
+	{
+		transform.localScale =
+		    transform.localScale.SetY( heightScaleRange.Random );
+	}
+
+	void AdjustColor()
+	{
+		Color colorOffset = new Color( -Mathf.Clamp01( colorOffsetRange.Random ),
+		                               Mathf.Clamp01( colorOffsetRange.Random ),
+		                               Mathf.Clamp01( colorOffsetRange.Random ),
```
Comment (brickman1444) [inline Assets/Scripts/Buddy/BuddyShaper.cs:63]: why are these clamped and why is the first one negated?

File: Assets/Scripts/Tags/NightLightTag.cs [hunk 7]
```diff
@@ -3,4 +3,57 @@
 
 public class NightLightTag : MonoBehaviour
 {
+	[Range( 0.0f, 1.0f )]
+	[SerializeField] float _timeOn = 0.6f;
+	[SerializeField] float _fadeTime = 1.0f;
```
Comment (brickman1444) [inline Assets/Scripts/Tags/NightLightTag.cs]: What even is this value? I can't tell by the name.

File: Assets/Scripts/Tags/NightLightTag.cs [hunk 8]
```diff
@@ -3,4 +3,57 @@
...
+
+	private Light _light;
+	private float _lightIntensity;
```
Comment (brickman1444) [inline Assets/Scripts/Tags/NightLightTag.cs]: This could be renamed to something like full intensity. Otherwise it's a bit confusing what's happening in the coroutine.

**Comment by brickman1444** [comment]
Done looking this over.

File: Assets/Scripts/Buddy/BuddyShaper.cs (hunk 6 above)
Comment (cjacobwade) [inline Assets/Scripts/Buddy/BuddyShaper.cs:63]: I was testing just on strength so that probably looks weird on blue/purple buddies. I use clamp because I'm pretty sure colors don't auto clamp and thus you get weird behavior if going above 1.

File: Assets/Scripts/Tags/NightLightTag.cs (hunk 7 above)
Comment (randomPoison) [inline Assets/Scripts/Tags/NightLightTag.cs]: The time it takes for the light to fade in. What would a more appropriate name be? `_fadeInTime`?

File: Assets/Scripts/Tags/NightLightTag.cs (hunk 8 above)
Comment (randomPoison) [inline Assets/Scripts/Tags/NightLightTag.cs]: Done.

File: Assets/Scripts/Tags/NightLightTag.cs (hunk 7 above)
Comment (brickman1444) [inline Assets/Scripts/Tags/NightLightTag.cs]: Yea that'd be better.
